modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2SynapseEnvironment.java

```diff
@@ -235,21 +236,30 @@ private void setSeverUnitTestMode(ServerContextInformation contextInformation) {
             //get unit test received port
             String requestPort = System.getenv("UNIT_TEST_PORT");
 
-            //set default port for TCP server connection
-            int unitTestingAgentPort = 9007;
-            if (requestPort != null && !requestPort.isEmpty()) {
-                try {
+            //check port for TCP server connection
+            int unitTestingAgentPort = -1;
+            try {
+                if (requestPort != null && !requestPort.isEmpty()) {
                     unitTestingAgentPort = Integer.parseInt(requestPort);
-                } catch (NumberFormatException e) {
-                    log.error("Given port is not in valid format", e);
+                } else {
+                    throw new IOException("Starting TCP port can not be empty or null");
                 }
+
+            } catch (NumberFormatException e) {
+                log.error("Given port is not in valid format", e);
+            } catch (IOException e) {
```

Comment:
Do we need to catch this exception and continue the logic . We don't have anything to continue thereafter isn't it.

Reply:
Changed the try block and moved the condition into the try block